jax/experimental/tree_math.py

```diff
@@ -131,13 +139,17 @@ def tree_flatten(self):
   def tree_unflatten(cls, _, args):
     return cls(*args)
 
-  def __len__(self):
+  @property
+  def size(self):
     values = tree_util.tree_leaves(self.tree)
     return sum(prod(jnp.shape(value)) for value in values)
```

Comment:
I am asking just out of curiosity; is there a specific reason why you prefer `prod(jnp.shape(v))` over `jnp.size`?

Reply:
No, they are entirely equivalent